lib/usecase/base/base_use_case.dart

```diff
@@ -17,3 +17,9 @@ abstract class NoParamsUseCase<T> extends BaseUseCase<Result<T>> {
 
   Future<Result<T>> call();
 }
+
+abstract class SimpleUseCase<T> {
```

Comment:
I think for better naming we should do some refactor here a bit like this

```dart
abstract class BaseUseCase<T extends Result> {
  const BaseUseCase();
}

abstract class FutureUseCase<T, P> extends BaseUseCase<Result<T>> {
  const UseCase() : super();

  Future<Result<T>> call(P params);
}

abstract class NoParamsFutureUseCase<T> extends BaseUseCase<Result<T>> {
  const NoParamsUseCase() : super();

  Future<Result<T>> call();
}

abstract class UseCase<T> {
  const UseCase();

  T call();
}
```

What do you think?

Reply:
@markgravity `UseCase` is most common used and the main usecase. Renaming it to `FutureUseCase` leads to a long base class name. I think I will keep the current one.